Formula/redo.rb

```diff
@@ -0,0 +1,38 @@
+class Redo < Formula
+  include Language::Python::Virtualenv
+
+  desc "Implements djb's redo: an alternative to make"
+  homepage "https://redo.rtfd.io/"
+  url "https://github.com/apenwarr/redo/archive/redo-0.42a.tar.gz"
+  sha256 "0ff0fd29f2767fe33374cfac97fd13b3dd27207bdacbeb5e02039b992d4fb815"
+
+  depends_on "python"
+
+  resource "Markdown" do
+    url "https://files.pythonhosted.org/packages/98/79/ce6984767cb9478e6818bd0994283db55c423d733cc62a88a3ffb8581e11/Markdown-3.2.1.tar.gz"
+    sha256 "90fee683eeabe1a92e149f7ba74e5ccdc81cd397bd6c516d93a8da0ef90b6902"
+  end
+
+  resource "beautifulsoup4" do
+    url "https://files.pythonhosted.org/packages/3b/e4/7cfc641f11e0eef60123912611a5c9ee7d4638da7325878b695b9ae4bb6f/beautifulsoup4-4.9.0.tar.gz"
+    sha256 "594ca51a10d2b3443cbac41214e12dbb2a1cd57e1a7344659849e2e20ba6a8d8"
+  end
+
+  def install
+    venv = virtualenv_create(libexec, "python3")
+    venv.pip_install resources
+    # Set the interpreter so that ./do install can find the pip installed
+    # resources
+    open("redo/whichpython", "w") do |file|
+      file.puts "#{libexec}/bin/python3"
+    end
```

Comment:
Is this the official way?

Reply:
I'm not entirely sure what you mean by "official" (_Homebrew_ official or _upstream_ official).

Reply:
Upstream official. While homebrew doesn't have an official way, this looks weird to me.

Reply:
I agree that this looks like a weird hack-y workaround. In my next commit, I'll comment out the offending lines. (It still works on my system without having those lines there, but it may not work with the Homebrew test bots. I haven't consulted with the upstream maintainers, but ideally, I would like something that changes `PATH` for that purpose.)